extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/linkHandler/YoutubeStreamLinkHandlerFactory.java

```diff
@@ -46,13 +48,25 @@ public static YoutubeStreamLinkHandlerFactory getInstance() {
         return instance;
     }
 
+    private static String extractId(@Nullable String id) {
+        if (id != null) {
+            Matcher m = Pattern.compile("([a-zA-Z0-9_-]{11})").matcher(id);
+            return m.find() ? m.group(1) : "";
```

Comment:
Return `null` instead of an empty string, if you found nothing (also remember to add `@Nullable` to the function)